juju/juju
# feat(firewall): wire up firewaller to use ports domain

Wire up the firewaller api on the controller facade to be backed by the DQLite ports domain.

This included substituting the DQLite watcher for the Mongo watcher.

This required a quick patch to the DQLite watcher to emit machine names instead of machine uuids, since this data is consumed outside the apiserver, so should be a natural key.

We also needed to wire up OpenedPortRangesForMachine. This was a little more complex.

The OpenedPortRangesForMachine spec requires us to return not just the endpoint with their opened ports, but the subnet cidrs that endpoint is opened to.

However, endpoint bindings have not yet been implemented in DQLite. So instead, I have created a new method on the service, GetMachineOpenedPortsAndSubnets, which for now just returns the opened ports with nil SubnetCIDRs. Once we have endpoint bindings wired up in DQLite, this stub method will need to be expanded.

This has been added to the risk register, and a ticket has been added to the appropriate epic for this work.

## Checklist


## QA steps

Run
```
juju bootstrap aws aws
juju add-model m
juju deploy ubuntu
juju expose ubuntu
juju exec --unit ubuntu/0 open-port 1000/tcp
```

Check debug logs for:
```
machine-0: 11:51:28 INFO juju.provider.ec2 opened ports in security group juju-0a8403c4-10cc-4b80-89d3-77ae26bb4b3d-0: [1000/tcp from 0.0.0.0/0,::/0]
```

verify the port 1000 is open on the machine sec group
![image](https://github.com/user-attachments/assets/49f1a578-417d-433f-ac65-4e60dc229443)
